### PR TITLE
Introduce seed in image classifier and stabilize unit test

### PR DESCRIPTION
**Pull request by gustavla**
The unit test `ImageClassifierTest::test_export_coreml_with_predict` failed with high probability. To address this, this PR increases the test tolerance and makes model creation optionally deterministic by exposing a seed parameter.

I changed the tolerance from 0.005 to the much higher 0.02. This is after running 25 trials and collecting some statistics. Only 3 times was it below 0.005 and the highest was around 0.015.

Note however that even though this tolerance seems awfully high, separate tests indicate that the error is many orders of magnitude smaller when the network is fed natural images (instead of randomly generated input as in this unit test).

### CODE REVIEW

File: src/unity/python/turicreate/toolkits/_supervised_learning.py [hunk 1]
```diff
@@ -295,7 +295,8 @@ def print_validation_track_notification():
 
 
 def create(dataset, target, model_name, features=None,
-           validation_set='auto', verbose=True, distributed='auto', **kwargs):
+           validation_set='auto', distributed='auto',
```
Comment (znation) [inline src/unity/python/turicreate/toolkits/_supervised_learning.py]: I think this is potentially a breaking API change, if callers were using positional arguments for `verbose` (which is now `distributed`) or `distributed` (which is now `seed`). Can we keep the order the same as before and add `seed` at the end of the positional arguments?

File: src/unity/python/turicreate/toolkits/classifier/logistic_classifier.py [hunk 2]
```diff
@@ -36,6 +36,7 @@ def create(dataset, target, features=None,
     max_iterations = _DEFAULT_SOLVER_OPTIONS['max_iterations'],
     class_weights = None,
     validation_set = 'auto',
+    seed=None,
```
Comment (znation) [inline src/unity/python/turicreate/toolkits/classifier/logistic_classifier.py]: Same comment here: can we preserve API compatibility?

File: src/unity/python/turicreate/toolkits/image_classifier/image_classifier.py [hunk 3]
```diff
@@ -27,7 +27,8 @@
 from turicreate.toolkits._internal_utils import (_raise_error_if_not_sframe,
                                                  _numeric_param_check_range)
 
-def create(dataset, target, feature = None, model = 'resnet-50', max_iterations = 10, verbose = True):
+def create(dataset, target, feature = None, model = 'resnet-50',
```
Comment (znation) [inline src/unity/python/turicreate/toolkits/image_classifier/image_classifier.py]: Same comment here.

Comment (gustavla) [reply src/unity/python/turicreate/toolkits/image_classifier/image_classifier.py]: This is a great point. I agree and will push a change.

Done.

File: src/unity/python/turicreate/toolkits/classifier/logistic_classifier.py (hunk 2 above)
Comment (gustavla) [reply src/unity/python/turicreate/toolkits/classifier/logistic_classifier.py]: Done.

File: src/unity/python/turicreate/toolkits/_supervised_learning.py (hunk 1 above)
Comment (gustavla) [reply src/unity/python/turicreate/toolkits/_supervised_learning.py]: Done.

**Pull request merged**